semattack/src/main_attack.cpp

```diff
@@ -41,25 +41,48 @@ std::string call_sem_attack(const std::string& target_name, const std::string& d
         DepGraph target_dep_graph;
         if (dep_graph != "") {
             target_dep_graph = DepGraph::parseString(dep_graph);
-        } else 
+        } else
         {
             target_dep_graph = DepGraph::parseDotFile(target_name);
         }
+
+        // Compute sink post image for target
         SemAttack semAttack(target_name, target_dep_graph, field_name);
         semAttack.setPrintDots(true);
         semAttack.init();
+
         AnalysisResult result = semAttack.computeTargetFWAnalysis();
         const StrangerAutomaton* postImage = semAttack.getPostImage(result);
-        StrangerAutomaton* attackPattern = AttackPatterns::getHtmlPattern();
-        StrangerAutomaton* intersection = semAttack.computeAttackPatternOverlap(postImage, attackPattern);
-// bw = doBackwardAnalysisForPayload(attr_payload, output_dir, computePreImage, singletonIntersection, outputDotfiles, attack_forward)
+
+        //cout << "Statsifying example of post Image " + postImage->generateSatisfyingExample();
+        //cout << "\n";
+
+        // Turn the exploit_string into an automaton
+        StrangerAutomaton* exploit = StrangerAutomaton::makeContainsString(exploit_string);
```

Comment:
Make sure you delete the pointers once you are finished with them - ie before returning.